contracts/src/core/BatcherPaymentService.sol

```diff
@@ -87,14 +91,28 @@ contract BatcherPaymentService is
     function initialize(
         IAlignedLayerServiceManager _alignedLayerServiceManager,
         address _batcherPaymentServiceOwner,
-        address _batcherWallet
+        address _batcherWallet,
+        bytes32 _noncedVerificationDataTypeHash
     ) public initializer {
         __Ownable_init(); // default is msg.sender
         __UUPSUpgradeable_init();
         _transferOwnership(_batcherPaymentServiceOwner);
 
         alignedLayerServiceManager = _alignedLayerServiceManager;
         batcherWallet = _batcherWallet;
+        noncedVerificationDataTypeHash = _noncedVerificationDataTypeHash;
+    }
+
+    function initializeNoncedVerificationDataTypeHash(
+        bytes32 _noncedVerificationDataTypeHash
+    ) public reinitializer(2) onlyOwner {
+        noncedVerificationDataTypeHash = _noncedVerificationDataTypeHash;
```

Comment:
```suggestion
        setNoncedVerificationDataTypeHash(_noncedVerificationDataTypeHash);
```

Reply:
I think it's more expensive and doesn't add any improvement in readability